### ISSUE TITLE
jpegtran don't rotate jpeg encoded with arithmetic coding

### ISSUE THREAD

**Issue by MihailZenkov** [mozilla/mozjpeg#174]
jpegtran don't rotate jpeg encoded with arithmetic coding. libjpeg-turbo don't have this problem.

**Comment by bumper314** [comment]
This bug doesn't seem to be limited to images with arithmetic coding.  I found that after optimizing, the resulting image will silently fail to rotate.

Test file (8mb): https://d.pr/i/SO9cX4

Output from `jpeginfo -i 142.jpg`:
```
142.jpg 4278 x 3056 24bit JFIF  Progressive Huffman 8221909
```

### Test 1
Rotate image with `jpegtran -outfile rotated.jpg -rotate 90 142.jpg`
=> Rotates successfully

### Test 2
Optimize with `jpegtran -copy none -progressive -outfile optimized.jpg 142.jpg`
Output from `jpeginfo -i optimized.jpg`:
```
optimized.jpg 4278 x 3056 24bit JFIF  Progressive Huffman 8114743
```

Now try rotating with `jpegtran -outfile notrotated.jpg -rotate 90 optimized.jpg`
=> Rotate failed, exit code of 0

### Test 3
Optimize and rotate at the same time works:
`jpegtran -outfile rotated.jpg -copy none -progressive -rotate 90 142.jpg`

**Comment by kornelski** [comment]
This is expected. It's because JPEG images can be losslessly rotated only if their size is even multiple of 8 or 16 (16 if the image uses chroma subsampling).

**Comment by bumper314** [comment]
I disagree.  Stock `jpegtran` works fine.  Why does the image rotate fine with the unoptimized image vs the failed rotate with the optimized, when they have the same dimensions?

Also, the rotate should only fail if the `-perfect` option is used.

**Comment by kkopachev** [comment]
This is fixed by #298.
